iodata/formats/mwfn.py

```diff
@@ -0,0 +1,315 @@
+    (5, 'c'): ['zzzzz', 'yzzzz', 'yyzzz', 'yyyzz', 'yyyyz', 'yyyyy', 'xzzzz', 'xyzzz',
+               'xyyzz', 'xyyyz', 'xyyyy', 'xxzzz', 'xxyzz', 'xxyyz', 'xxyyy', 'xxxzz',
+               'xxxyz', 'xxxyy', 'xxxxz', 'xxxxy', 'xxxxx'],
+}
+
+
+def _load_helper_opener(lit: LineIterator) -> dict:
+    """Read initial variables at the beginning of a MWFN file."""
+    keys = {"Wfntype": int, "Charge": float, "Naelec": float, "Nbelec": float, "E_tot": float,
+            "VT_ratio": float, "Ncenter": int}
+    max_count = len(keys)
+    count = 0
+    data = {}
+    while count < max_count:
+        line = next(lit)
+        for name, ftype in keys.items():
+            if name in line:
+                data[name] = ftype(line.split('=')[1].strip())
+                count += 1
+
+    # check values parsed
+    if data["Ncenter"] <= 0:
+        lit.error(f"Ncenter should be a positive integer! Read Ncenter= {data['Ncenter']}")
+    # Possible values of Wfntype (wavefunction type):
+    #     0: Restricted closed - shell single - determinant wavefunction(e.g.RHF, RKS)
+    #     1: Unrestricted open - shell single - determinant wavefunction(e.g.UHF, UKS)
+    #     2: Restricted open - shell single - determinant wavefunction(e.g.ROHF, ROKS)
+    #     3: Restricted multiconfiguration wavefunction(e.g.RMP2, RCCSD)
+    #     4: Unrestricted multiconfiguration wavefunction(e.g.UMP2, UCCSD)
+    if data["Wfntype"] in [0, 2, 3]:
+        # restricted wavefunction
+        data["mo_kind"] = "restricted"
+    elif data["Wfntype"] in [1, 4]:
+        # unrestricted wavefunction
+        data["mo_kind"] = "unrestricted"
+    else:
+        lit.error(f"Wavefunction type cannot be determined. Read Wfntype= {data['Wfntype']}")
+
+    return data
+
+
+def _load_helper_basis(lit: LineIterator) -> dict:
+    """Read basis set information typically labelled '# Basis function information'."""
+    # Nprims must be last or else it gets read in with Nprimshell
+    keys = ["Nbasis", "Nindbasis", "Nshell", "Nprimshell", "Nprims"]
+    count = 0
+    data = {}
+    next(lit)
+    while count < len(keys):
+        line = next(lit)
+        for name in keys:
+            if name in line:
+                data[name] = int(line.split('=')[1].strip())
+                count += 1
+                break
+    return data
+
+
+def _load_helper_atoms(lit: LineIterator, natom: int) -> dict:
+    """Read atoms section typically labelled '# Atom information'."""
+    data = {
+        "atnums": np.empty(natom, int),
+        "atcorenums": np.empty(natom, float),
+        "atcoords": np.empty((natom, 3), float),
+    }
+
+    # skip lines until "$Centers" section is reached
+    line = next(lit)
+    while '$Centers' not in line and line is not None:
+        line = next(lit)
+
+    for atom in range(natom):
+        words = next(lit).split()
+        data["atnums"][atom] = int(words[2].strip())
+        data["atcorenums"][atom] = float(words[3].strip())
```

Comment:
very minor:
```suggestion
        data["atnums"][atom] = words[2]
        data["atcorenums"][atom] = words[3]
```
Split returns stripped results and numpy does conversion from string upon assignment. E.g.

```python
>>> import numpy as np
>>> a = np.zeros(3)
>>> a[1] = "5.0"
>>> a
array([0., 5., 0.])
>>> b = np.zeros(3, int)
>>> b[0] = "134 "
>>> b
array([134,   0,   0])
```